### ISSUE TITLE
Add "metadata" to bucket model for handling additional info when querying branches

### ISSUE THREAD

**Issue by vasilievip** [cdancy/bitbucket-rest#71]
If you provide details=true when you hit following link:
http://<bitbucket>/rest/api/1.0/projects/<project>/repos/<repo>/branches?details=true

branch information has additional details under "metadata" key, it would be great to access it in java

**Comment by cdancy** [comment]
@vasilievip looking at their[ REST API I see that the details](https://developer.atlassian.com/static/rest/bitbucket-server/4.2.1/bitbucket-rest.html#idp2897856) flag brings in additional info about `plugin-provided metadata about each branch`. Is that what you're looking for? Care to put together a PullRequest or were you looking for one of us to do so? Let me know. Thanks.

**Comment by cdancy** [comment]
@vasilievip jumped the gun a bit, looks like you were looking to add this to `branchModel` endpoint. Do you have a link to the documentation somewhere? Can't seem to find it at the moment...

**Comment by vasilievip** [comment]
I can't find  link to documentation, but I can give you example which I see when I hit endpoint.
(i'll try to come up with pull request a bit later). What I see in metadata is very custom items produced by plugins, I guess handling it as a String or even Map datatype would be OK. 
```
{
  "size": 1,
  "limit": 1,
  "isLastPage": false,
  "values": [
    {
      "id": "refs/heads/feature/sticky-fix",
      "displayId": "feature/sticky-fix",
      "type": "BRANCH",
      "latestCommit": "026e014dc71376bafc601c155fca75bd5dab8205",
      "latestChangeset": "026e014dc71376bafc601c155fca75bd5dab8205",
      "isDefault": false,
      "metadata": {
        "com.atlassian.bitbucket.server.bitbucket-jira:branch-list-jira-issues": [],
        "com.atlassian.bitbucket.server.bitbucket-branch:ahead-behind-metadata-provider": {
          "ahead": 1,
          "behind": 0
        },
        "com.atlassian.bitbucket.server.bitbucket-branch:latest-commit-metadata": {
          "id": "026e014dc71376bafc601c155fca75bd5dab8205",
          "displayId": "026e014dc71",
          "author": {
            "name": "Sergey",
            "emailAddress": "slysyuk@acme.com",
            "avatarUrl": "https://secure.gravatar.com/avatar/724799292f3a225600f4d4644468171c.jpg?s=64&d=mm"
          },
          "authorTimestamp": 1495834461000,
          "message": "sticky wild fix",
          "parents": [
            {
              "id": "5114f8c7dcc51f9c85a223429938a0b5598d238d",
              "displayId": "5114f8c7dcc"
            }
          ]
        },
        "com.atlassian.bitbucket.server.bitbucket-build:build-status-metadata": {
          "successful": 1,
          "inProgress": 0,
          "failed": 0
        }
      }
    }
  ],
  "start": 0,
  "nextPageStart": 1
}
```

**Comment by vasilievip** [comment]
https://github.com/cdancy/bitbucket-rest/pull/72

**Comment by cdancy** [comment]
Issue is resolved per PR #80
